src/api/handlers/project.py

```diff
@@ -364,12 +373,18 @@ def post(self, project_id):
                 VALUES (null, %s, %s, %s, %s)
             ''', [build_number, project_id, source_upload_id, build_id])
 
+            # create job matrix should be rescheduled if HA enabled
+            if os.environ['INFRABOX_HA_ENABLED'] == 'true':
+                job_cluster_name = None
+            else:
+                job_cluster_name = "master"
+
             g.db.execute('''
                 INSERT INTO job (id, state, build_id, type, name, project_id,
-                                 dockerfile, build_only, cpu, memory)
+                                 dockerfile, build_only, cpu, memory, cluster_name)
```

Comment:
it's the schedulers job to chose a cluster. cluster_name should always be false here.

Reply:
scheme cluster_name defination is   `cluster_name     varchar default 'master' :: character varying`.

Set cluster_name here is to stop the job from scheduling to master.

May be we should set cluster_name to null explictly.